cmake/modules/platform/toolcfg/openxl.cmake

```diff
@@ -0,0 +1,145 @@
+###############################################################################
+# Copyright IBM Corp. and others 2024
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which accompanies this
+# distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+# or the Apache License, Version 2.0 which accompanies this distribution
+# and is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception [1] and GNU General Public
+# License, version 2 with the OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] https://openjdk.org/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+###############################################################################
+
+
+if(OMR_OS_ZOS)
+	set(OMR_ZOS_COMPILE_ARCHITECTURE "arch10" CACHE STRING "z/OS compile machine architecture" FORCE)
+	set(OMR_ZOS_COMPILE_TARGET "ZOSV2R4" CACHE STRING "z/OS compile target operating system" FORCE)
+	set(OMR_ZOS_COMPILE_TUNE "12" CACHE STRING "z/OS compile machine architecture tuning" FORCE)
+	set(OMR_ZOS_LINK_COMPAT "ZOSV2R4" CACHE STRING "z/OS link compatible operating system" FORCE)
+	set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "--shared")
+	set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "--shared")
+
+	set(CMAKE_ASM_FLAGS "-fno-integrated-as")
+	string(APPEND CMAKE_ASM_FLAGS " \"-Wa,-mgoff\"")
+	if(OMR_ENV_DATA64)
+		string(APPEND CMAKE_ASM_FLAGS " \"-Wa,-mSYSPARM(BIT64)\"")
+	endif()
+
+	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+		"-fstrict-aliasing"
+		"-mzos-target=${OMR_ZOS_COMPILE_TARGET}"
+		"-m64"
+	)
+
+	list(APPEND OMR_PLATFORM_C_COMPILE_OPTIONS
+		-march=${OMR_ZOS_COMPILE_ARCHITECTURE}
+	)
+
+	list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS
+		-march=${OMR_ZOS_COMPILE_ARCHITECTURE}
+		"-std=gnu++14"
```

Comment:
Why `gnu` and not `c` as in `-std=c++14`?

Reply:
Sean from the cpp team recommended to use `gnu++14` over `c++14` to avoid using `_EXT` in most places ...
More specifics regarding feedback there:
`_EXT` required for alloca() not being exposed in `javadump.cpp`
```
Use -std=gnu++14.  The closest XL option to -std=c++14 is -qlanglvl=ansi.  With that XL option you would need to add 
_EXT too.  The -std=gnu* options are the closest to -qlanglvl=etended.   stdlib.h will declare alloca() with -std=gnu*.
---
If you say you want a standard language level (like std=c++14), then you do need to specify the _EXT to say you want
extensions. If you use extension based language levels (like the gnu versions) then you get most extensions enabled by
default.
```

Reply:
Ok. Please add a comment that summarizes that.

Reply:
I don't see that comment.